### PR TITLE
Add abstract route to span name https://github.com/openzipkin/zipkin-ruby/issues/96

### PR DESCRIPTION
**Pull request by jmhon08**
screen shot 2018-06-20 at 10 19 41 am

### CODE REVIEW

**Comment by jmhon08** [comment]
@ykitamura-mdsol @jcarres-mdsol could you please take a look?

File: lib/zipkin-tracer/rack/zipkin-tracer.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def call(env)
         if !trace_id.sampled? || !routable_request?(env)
           @app.call(env)
         else
-          @tracer.with_new_span(trace_id, env[REQUEST_METHOD].to_s.downcase) do |span|
+          @tracer.with_new_span(trace_id, "#{env[REQUEST_METHOD].to_s.downcase} #{route(env)}".strip) do |span|
```
Comment (adriancole) [inline lib/zipkin-tracer/rack/zipkin-tracer.rb]: maybe put in a unit test for this, as we don't want a span name of "get nil" or "get " If there's no route, the method alone should be the name.

File: spec/lib/rack/zipkin-tracer_spec.rb [hunk 2]
```diff
@@ -75,30 +79,48 @@ def expect_host(host)
   context 'Using Rails' do
     subject { middleware(app) }
 
-    context 'accessing a valid URL of our service' do
+    context 'accessing a valid URL "/" of our service' do
       before do
-        allow(middleware(app)).to receive(:routable_request?).and_return(true)
```
Comment (jmhon08) [inline spec/lib/rack/zipkin-tracer_spec.rb]: Previously, if I changed this to and_return(false), I would still see routable_request?(env) return true in rack/zipkin-tracer.rb

File: lib/zipkin-tracer/rack/zipkin-tracer.rb (hunk 1 above)
Comment (jmhon08) [reply lib/zipkin-tracer/rack/zipkin-tracer.rb]: Done in latest commit, can you please review?

**Comment by jmhon08** [comment]
@adriancole @ykitamura-mdsol @jcarres-mdsol could someone pls merge this?

**Comment by adriancole** [comment]
thanks! while not a rails expert, the change you made wrt route looks sound. folks can feel free to give more feedback post merge.

**Pull request merged**